site-src/geps/gep-1282.md

```diff
@@ -98,12 +98,302 @@ Of course, from the section above, you can see that there are many other feature
+```go
+// BackendCapabilities references a Kubernetes Service and defines any backend 
+// capabilities that Service requires.  
+//
+// +optional
+type BackendCapabilities struct {
+    // BackendName is the name of the backend service to which these backend 
+    // capabilities apply.
+    // +required
+    BackendName string `json:”backendName”`
+
+    // BackendType is the type of backend to which these backend 
+    // capabilities apply.
+    //
+    // +required
+    BackendType BackendTargetRef `json:”backendType”`
+
+	// Spec specifies the capabilities supported by this
+    // Service wrapper, if any.  The capabilities are predefined, and have two
+    // characteristics in common: they are tightly bound to the backend, 
+    // rather than being important at the route level, and the service owner
+    // (application developer in the Gateway API personas) should control 
+    // these settings, not the owner of the Gateway implementation (the 
+    // Cluster Operator in the Gateway API personas).
+    // 
+    // Support: Extended
+	//
+    // +required
+	Spec BackendCapabilitiesSpec `json:”spec”`
+
+	// Status represents the current state of the backend and its capabilities.
+    // +optional
+	Status *BackendCapabilitiesStatus `json:”status,omitempty”`
+}
+
+// BackendTargetReference identifies an API object to apply BackendCapabilities 
+// to. 
+// 
+type BackendTargetReference struct {
+	// Group is the group of the target resource.
+	Group Group `json:"group"`
+
+	// Kind is kind of the target resource.
+    // Valid values include:
+    // * "Service" (the default if no Kind is specified)
+    // * "ServiceImport"
+    //
+    // Invalid values include:
+    //
+    // * "invalid/kind" - "/" is an invalid character
+    // +required
+	Kind string `json:"kind"`
+
+	// Version is the version of the target resource.
+	Version ObjectName `json:"version"`
+
+	// Name is the name of the target resource.
+	Name ObjectName `json:"name"`
+}
+
+// BackendCapabilitiesSpec defines a group of specific properties held by its
+// affiliated BackendCapabilitiesService, and specified by the Service owner 
+// (typically the application developer).  Examples of properties that 
+// belong in the BackendCapabilitiesSpec include: TLS re-encryption, session
+// state, timeouts, retries, Websockets, HTTP/2, load balancer algorithm, 
+// healthcheck info, external traffic policy, connection limits, error page,
+// protocol switch, alerts, and others. 
+//
+// Support: Extended
+//
+type BackendCapabilitiesSpec struct {
+
+	// +optional
+	Reencrypt *TLSReencryptType `json:”reencrypt,omitempty”`
+
+	// +optional
+    Websocket *WebsocketType `json:”websocket,omitempty”`
+
+	// +optional
+    Protocol *ProtocolType `json:”protocol,omitempty”`
+}
+
+type ReencryptType struct {
+    Version *TLSVersion
+    ClientCertFile string
+    PrivateKeyFile string
+    BackendCertFile string
```

Comment:
What are these for? It may be helpful to see an e2e use case where all of these fields were used with an example of who populated each of these. As a broader note, these feel like something we should derive from Secret/Object refs, not file paths.

Reply:
I'll make some changes on this, using `SecretObjectReference` from https://github.com/kubernetes-sigs/gateway-api/blob/96414f1ce72b2277cde5cb8e8abcd97ea039fe4a/apis/v1beta1/object_reference_types.go#L48  
```suggestion
 type ReencryptType struct {
    // +optional	
    MinVersion   *TLSVersion
    // +optional    
    MaxVersion   *TLSVersion
    // +optional	
    ClientCerts  []SecretObjectReference
    // +optional	
    BackendCerts []SecretObjectReference
    // +optional	
    CaCerts	     []SecretObjectReference
}
```

Reply:
I changed it again.